nipoppy/workflows/dataset_init.py

```diff
@@ -85,13 +107,65 @@ def run_main(self):
             FPATH_SAMPLE_MANIFEST, self.layout.fpath_manifest, log_level=logging.DEBUG
         )
 
+        if self.bids_source is not None:
+            self._init_manifest_from_bids_dataset()
+
         # inform user to edit the sample files
         self.logger.warning(
             f"Sample config and manifest files copied to {self.layout.fpath_config}"
             f" and {self.layout.fpath_manifest} respectively. They should be edited"
             " to match your dataset"
         )
 
+    def _init_manifest_from_bids_dataset(self) -> None:
+        """Assume a BIDS dataset with session level folders.
+
+        No BIDS validation is done.
+        """
+        df = {"participant_id": [], "visit_id": [], "session_id": [], "datatype": []}
+        participant_ids = sorted(
+            [
+                x.name
+                for x in (self.layout.dpath_bids).iterdir()
+                if x.is_dir() and x.name.startswith("sub-")
+            ]
+        )
+
+        self.logger.info("Creating a manifest.csv from the BIDS dataset content.")
```

Comment:
```suggestion
        self.logger.info("Creating a manifest file from the BIDS dataset content.")
```
since we will change the manifest and other files to be TSV sometime in the future